include/onnxruntime/core/optimizer/rewrite_rule.h

```diff
@@ -35,6 +35,17 @@ If the list of op types is left empty, that rule will be triggered for every op
 */
 class RewriteRule {
  public:
+  /**
+  @class RewriteRuleEffect
+
+  Class used to indicate the effect of rule application on a graph's node.
+  */
+  enum class RewriteRuleEffect : uint8_t {
```

Comment:
>uint8_t [](start = 33, length = 7)

As this isn't stored anywhere, why use an 8 bit value vs the default?

Reply:
I left it as is to make sure the compiler picks the narrowest type. It will not help much in most cases but for sure it will not harm to have it.